Mathlib/NumberTheory/LSeries/MellinEqDirichlet.lean

```diff
@@ -0,0 +1,154 @@
+    rw [mul_cpow_ofReal_nonneg hpi.le ht.le, ← mul_assoc, one_div, inv_mul_cancel, one_mul]
+    · rw [Ne.def, cpow_eq_zero_iff, not_and_or]
+      exact Or.inl (ofReal_ne_zero.mpr hpi.ne')
+  · -- summability of integrals of norms
+    apply Summable.of_norm
+    convert h_sum.mul_left (Real.Gamma s.re) using 2 with i
+    simp_rw [← mul_assoc, mul_comm _ (a i), mul_assoc, norm_mul (a i), integral_mul_left]
+    rw [← mul_div_assoc, mul_comm (Real.Gamma _), mul_div_assoc, norm_mul ‖a i‖, norm_norm]
+    rcases hp i with hai | hpi
+    · simp only [hai, norm_zero, zero_mul]
+    congr 1
+    have := Real.integral_rpow_mul_exp_neg_mul_Ioi hs hpi
+    simp_rw [← neg_mul (p i), one_div, inv_rpow hpi.le, ← div_eq_inv_mul] at this
+    rw [norm_of_nonneg (integral_nonneg (fun _ ↦ norm_nonneg _)), ← this]
+    refine set_integral_congr measurableSet_Ioi (fun t ht ↦ ?_)
+    rw [norm_mul, norm_real, Real.norm_eq_abs, Real.abs_exp, Complex.norm_eq_abs,
+      abs_cpow_eq_rpow_re_of_pos ht, sub_re, one_re]
+
+/-- Shortcut version for the commonly arising special case when `p i = π * q i` for some other
+sequence `q`. -/
+lemma hasSum_mellin_pi_mul {a : ι → ℂ} {q : ι → ℝ} {F : ℝ → ℂ} {s : ℂ}
+    (hq : ∀ i, a i = 0 ∨ 0 < q i) (hs : 0 < s.re)
+    (hF : ∀ t ∈ Ioi 0, HasSum (fun i ↦ a i * rexp (-π * q i * t)) (F t))
+    (h_sum : Summable fun i ↦ ‖a i‖ / (q i) ^ s.re) :
+    HasSum (fun i ↦ π ^ (-s) * Gamma s * a i / q i ^ s) (mellin F s) := by
+  have hp i : a i = 0 ∨ 0 < π * q i := by rcases hq i with h | h <;> simp [h, pi_pos]
+  convert hasSum_mellin hp hs (by simpa using hF) ?_ using 2 with i
+  · have : a i / ↑(π * q i) ^ s = π ^ (-s) * a i / q i ^ s := by
+      rcases hq i with h | h
+      · simp [h]
+      · rw [ofReal_mul, mul_cpow_ofReal_nonneg pi_pos.le h.le, ← div_div, cpow_neg,
+          ← div_eq_inv_mul]
+    simp_rw [mul_div_assoc, this]
+    ring_nf
+  · have (i : ι) : ‖a i‖ / ↑(π * q i) ^ s.re = π ^ (-s.re) * ‖a i‖ / q i ^ s.re := by
+      rcases hq i with h | h
+      · simp [h]
+      rw [mul_rpow pi_pos.le h.le, ← div_div, rpow_neg pi_pos.le, ← div_eq_inv_mul]
+    simpa only [this, mul_div_assoc] using h_sum.mul_left _
+
+/-- Version allowing some constant terms (which are omitted from the sums). -/
+lemma hasSum_mellin_pi_mul₀ {a : ι → ℂ} {p : ι → ℝ} {F : ℝ → ℂ} {s : ℂ}
+    (hp : ∀ i, 0 ≤ p i) (hs : 0 < s.re)
+    (hF : ∀ t ∈ Ioi 0, HasSum (fun i ↦ if p i = 0 then 0 else a i * rexp (-π * p i * t)) (F t))
+    (h_sum : Summable fun i ↦ ‖a i‖ / (p i) ^ s.re) :
+    HasSum (fun i ↦ π ^ (-s) * Gamma s * a i / p i ^ s) (mellin F s) := by
+  have hs' : s ≠ 0 := fun h ↦ lt_irrefl _ (zero_re ▸ h ▸ hs)
+  let a' i := if p i = 0 then 0 else a i
+  have hp' i : a' i = 0 ∨ 0 < p i := by
+    simp only [a']
+    split_ifs with h <;> tauto
+    exact Or.inr (lt_of_le_of_ne (hp i) (Ne.symm h))
+  have (i t) : (if p i = 0 then 0 else a i * rexp (-π * p i * t)) =
+    (a' i * rexp (-π * p i * t)) := by simp only [a', ite_mul, zero_mul]
```

Comment:
```suggestion
  have (i t) : (if p i = 0 then 0 else a i * rexp (-π * p i * t)) =
      (a' i * rexp (-π * p i * t)) := by
    simp only [a', ite_mul, zero_mul]
```
I think it is better to have the proof on the next line when the statement spans more than one line.